components/flash_memory/flash_nvs.c

```diff
@@ -1,14 +1,122 @@
 // Copyright 2022 PWr in Space, Krzysztof Gliwiński
+// Edited Mateusz Kłosiński 21.02.2024
 #include "flash_nvs.h"
 
 #define TAG "NVS"
 
 static struct {
-  nvs_handle_t handle;
-  bool initialized;
+    nvs_handle_t handle;
+    bool initialized;
 } nvs;
 
-NVSResult NVS_init(void) {
+nvs_handle_t handle;
+/**
+ * #include "biblioteka_nvs.h"
+
+#define KEY_1 "1"
+#define KEY_1_VALUE 1
+#define KEY_2 "2"
+#define KEY_2_VALUE 2
+#define KEY_3 "3"
+#define KEY_3_VALUE 3
+
+void app_main(void) {
+    ESP_LOGI(TAG, "INIT TASK");
+
+    nvs_mem_t key_val_pairs[] = {
+        {KEY_1, KEY_1_VALUE},
+        {KEY_2, KEY_2_VALUE},
+        {KEY_3, KEY_3_VALUE}
+    };
+
+    nvs_init(key_val_pairs, sizeof(key_val_pairs) / sizeof(key_val_pairs[0]));
+    nvs_write(KEY_1, 321);
+
+    uint32_t value;
+    nvs_read(KEY_2, &value);
+}
+ *
+ */
+
+NVSResult NVS_init(NVSData* data_array, uint8_t length) {
+    esp_err_t res;
+
+    res = nvs_flash_init();
+    if (res != ESP_OK) {
+        ESP_LOGE(TAG, "NVS error init %s", esp_err_to_name(res));
+        return NVS_INIT_ERROR;
+    }
+
+    nvs.handle = 0;
+
+    if (length == 0 || data_array == NULL) {
+        length = 0;
+        ESP_LOGE(TAG, "data array is length 0 %s", esp_err_to_name(res));
+        return NVS_INIT_ERROR;
+    }
+
+    uint32_t tmp_Val;
+    for (int i = 0; i < length; i++) {
+        NVSResult result = NVS_read_uint32t(data_array[i].key, &tmp_Val);
+        if (result == NVS_OK && tmp_Val==data_array[i].value) {
+            ESP_LOGI(TAG, "Key %s already exists in NVS, skipping", data_array[i].key);
+        }
+        else if (result == NVS_OK && tmp_Val!=data_array[i].value) {
```

Comment:
Nie o to chodziło. Jeżeli klucz istnieje to nie chcemy aktualizować wartości, jeżeli nie to tworzymy klucz w pamięci i aktualizujemy wartość. Nie musimy sprawdzać jaka jest tam wartość. Wcześniej chyba było dobrze :nerd_face:

Reply:
No tutaj zrozumiałem tak że mamy sprawdzić czy value naszego klucza jest takie samo, jak jest takie samo to nie zapisujemy tego samego żeby oszczedzic flasha

Reply:
Ale to nie w init tylko w NVS_write_int32